gh-132661: Document t-strings and `templatelib` (#135229)
python/cpython: Doc/library/string.rst

The :meth:`str.format` method and the :class:`Formatter` class share the same
syntax for format strings (although in the case of :class:`Formatter`,
subclasses can define their own format string syntax).  The syntax is
related to that of :ref:`formatted string literals <f-strings>` and
:ref:`template string literals <t-strings>`, but it is less sophisticated

> and, in particular, does not support arbitrary expressions.

Reviewer: ```suggestion
and, in particular, does not support arbitrary expressions in interpolations.
```

.. _template-strings-pep292:

Template strings ($-strings)
----------------------------

.. note::

   The feature described here was introduced in Python 2.4.  It is unrelated
   to, and should not be confused with, the newer

>    :ref:`template strings <template-strings>` feature and

Reviewer 2: ```suggestion
   :ref:`template strings <template-strings>` and
```

> .. _template-strings-pep292:

Reviewer: This will break anchor references within the page: https://docs.python.org/3/library/string.html#template-strings

Are we OK with doing so?

Template strings ($-strings)
----------------------------

.. note::

>    The feature described here was introduced in Python 2.4.  It is unrelated
>    to, and should not be confused with, the newer
>    :ref:`template strings <template-strings>` and
>    :ref:`t-string literal syntax <t-strings>` introduced in Python 3.14.
>    T-string literals evaluate to instances of a different
>    :class:`~string.templatelib.Template` class, found in the
>    :mod:`string.templatelib` module.

Reviewer: ```suggestion
   The feature described here was introduced in Python 2.4
   as a simple templating method based upon regular expressions.
   It predates :meth:`str.format`, :ref:`formatted string literals <f-strings>`,
   and :ref:`template string literals <template-strings>`.

   It is unrelated to template string literals (t-strings),
   which were introduced in Python 3.14.
   These evaluate to  :class:`string.templatelib.Template` objects,
   found in the :mod:`string.templatelib` module.
```
